src/control/cmd/daos_agent/attachinfo.go

```diff
@@ -20,8 +20,9 @@ import (
 type dumpAttachInfoCmd struct {
 	configCmd
 	ctlInvokerCmd
-	Output string `short:"o" long:"output" default:"stdout" description:"Dump output to this location"`
-	JSON   bool   `short:"j" long:"json" description:"Enable JSON output"`
+	Output      string `short:"o" long:"output" default:"stdout" description:"Dump output to this location"`
+	JSON        bool   `short:"j" long:"json" description:"Enable JSON output"`
+	ProviderIdx uint   `short:"n" long:"provider_idx" description:"Index of provider to fetch (if multiple)"`
```

Comment:
I know this command is mostly used by developers/test stuff, but I'm wondering how the user would know which index to supply.

Reply:
The admin would have access to the server config file. In the multiprovider case, the admin will have defined something like: `provider: ofi+verbs,ofi+tcp`. Thinking of it as a list, the 0-based index corresponds to items in this list.

Reply:
Also: I'm planning to shift the agent config to use the provider index to specify which one to use as well, instead of the name. So it will be a bit more consistent. I can't think of a better way at the moment for the admin to specify which provider to use if the primary and secondaries are the same, just using different networks.

Reply:
Hmm. I may be overthinking it, but having to look at the server config in order to determine the correct index doesn't seem like good UX to me. Is there another command that the admin could run on the agent to see the list of indexed providers? Maybe the ROI isn't there to justify adding it, dunno. I don't know how often non-developers will need to use this command.